### DISCUSSION TITLE
cfclient not working in windows 11 after fresh installation

### DISCUSSION THREAD

**Discussion by AT882** [bitcraze/discussions#1332] [Q&A]
Hello all 
cfclient not working in windows after fresh installation.its showing error as in the screenshot attached (versions of python and git installed is also shown)
![Screenshot 2024-05-08 160533](https://github.com/bitcraze/discussions/assets/141340446/54a25514-7c8f-4a93-a7d1-c2aeb67c0f42)

Any help is appreciated

**Comment by ToveRumar** [comment]
Hi!

This looks like there could  some issue with your python environment. 
It looks like you are not running in a virtual environment, is that correct?
Could you check your PyQT version?  
Did you install from source or via pip?
Did every python packet install correctly when you where installing them?

**Reply by AT882** [reply]
yes i am not running it in a virtual environment, PyQT version is shown in the screenshot attached below
i had installed it from the source and yes every phyton packet was installed correctly it didn't show any errors during installation
![Screenshot 2024-05-14 095228](https://github.com/bitcraze/discussions/assets/141340446/8c78ffc7-e5da-4de8-b827-e9398770252b)

**Reply by knmcguire** [reply]
Hi! I'm able to run cfclient fine on my windows 11 with a python3.10 install, so sadly I can't recreate the same issue, but perhaps I should try it with 3.12 as well.

There is perhaps some misunderstanding as well. Your pip package shows that you installed the cfclient from release, not source. From source means downloading the githubrepo and doing `pip3 install -e .` Take a look at the installation instructions: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/installation/install/

Did you try just running `cfclient`, not `cfclient.gui`?

**Reply by AT882** [reply]
i have now downgraded python to 3.10 still the same issue exists
![Screenshot 2024-05-15 173201](https://github.com/bitcraze/discussions/assets/141340446/955efd69-2306-45f2-8c1e-3e7e2b78bdb7)
if possible can you share a video tutorial of the whole process.

**Reply by knmcguire** [reply]
I don't think a video tutorial will be suitable. The guide I send earlier is the only official installation documentation, but I can give a step by step.

What I've done is to start up a new python environment with 3.10 and activated it (to make sure that it's a fresh install)

```
mkdir venv_cfclient
python -m venv venv_cfclient
.\venv_test3\Scripts\activate
```

Upgrade pip and check pip list

```
pip3 install --upgrade pip
pip3 list
```

This showed:
```
Package             Version
------------------- --------
appdirs             1.4.4
cfclient            2024.3
cflib               0.1.25.1
freetype-py         2.4.0
hsluv               5.0.4
importlib_resources 6.4.0
kiwisolver          1.4.5
libusb-package      1.0.26.2
numpy               1.26.4
packaging           24.0
pip                 23.0.1
PyOpenGL            3.1.7
PyQt6               6.7.0
PyQt6-Qt6           6.7.0
PyQt6-sip           13.6.0
pyqtgraph           0.13.7
PySDL2              0.9.16
pysdl2-dll          2.24.0
pyserial            3.5
pyusb               1.2.1
PyYAML              6.0.1
pyzmq               25.1.2
scipy               1.13.0
setuptools          65.5.0
vispy               0.14.2
```

Then CFclient startup

```
 cfclient
UserWarning: Using SDL2 binaries from pysdl2-dll 2.24.0
INFO:cfclient.gui:Disabling STL printouts
INFO:cfclient.utils.input.inputreaders:Input readers: ['linuxjsdev', 'pysdl2']
INFO:cfclient.utils.input.inputreaders:Could not initialize [linuxjsdev]: Only supported on Linux
INFO:cfclient.utils.input.inputreaders:Successfully initialized [pysdl2]
INFO:cfclient.utils.input.inputinterfaces:Found interfaces: ['leapmotion', 'wiimote', 'zmqpull']
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [leapmotion]: Leap Motion library probably not installed (No module named 'leapsdk')
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [wiimote]: Missing cwiid (wiimote) driver No module named 'cwiid'
INFO:cfclient.utils.config:Dist config read from C:\Users\kimbe\Development\bitcraze\python\venv_test3\lib\site-packages\cfclient/configs/config.json
INFO:cfclient.utils.config:Config file read from [C:\Users\kimbe\AppData\Local\Bitcraze\cfclient/config.json]
INFO:cfclient.utils.input.inputinterfaces:Could not initialize [zmqpull]: ZMQ input disabled in config file
INFO:OpenGL.acceleratesupport:No OpenGL_accelerate module loaded: No module named 'OpenGL_accelerate'
INFO:cfclient.utils.zmq_led_driver:ZMQ led disabled in config file
INFO:cfclient.utils.zmq_param:ZMQ param disabled in config file
INFO:cfclient.utils.input:Using device blacklist [(VirtualBox|VMware)]
Cannot find a Crazyradio Dongle
INFO:cflib.drivers.cfusb:Looking for devices....
INFO:cflib.crtp.prrtdriver:Initialized PRRT driver.
INFO:cfclient.utils.input.inputreaders.pysdl2:Looking for devices
INFO:cfclient.utils.input.inputreaders.pysdl2:Found 0 devices
INFO:cfclient.utils.input.inputreaders.pysdl2:Looking for devices
INFO:cfclient.utils.input.inputreaders.pysdl2:Found 0 devices
INFO:cfclient.utils.input.inputreaders.pysdl2:Looking for devices
INFO:cfclient.utils.input.inputreaders.pysdl2:Found 0 devices
```

It's working for me.

I see that your QT version are a bit outdated. Since the error comes from that that might be the curlprit.

Make sure to first prepare your windows with these instructions: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/installation/install/#windows before installing it with pip via these instructions: https://www.bitcraze.io/documentation/repository/crazyflie-clients-python/master/installation/install/#from-pypi-windows-mac-linux--with-python3

**Comment by AT882** [comment]
I have tried cfclient on virtual box with the ova file that your side has provided. right now its working fine via virtual machine
Thank you all for the help.

**Reply by knmcguire** [reply]
alright, thanks for letting us know. It would have been great to figure out why your cfclient on windows didn't work but at least you are unblocked
